cpp/tests/rolling/range_window_bounds_test.cpp

```diff
@@ -0,0 +1,113 @@
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/cudf_gtest.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <thrust/iterator/constant_iterator.h>
+#include <thrust/iterator/counting_iterator.h>
+
+#include <chrono>
+#include <cuda/std/ratio>
+#include <vector>
+
+// TODO: Add to rolling.hpp.
+#include <cudf/rolling/range_window_bounds.hpp>
+#include <src/rolling/range_window_bounds_detail.hpp>
```

Comment:
I object to put the `*.hpp` files to the `src/` folder. If we don't want to expose the APIs in a hpp then we can put it to a corresponding `detail` folder inside the `include/cudf` folder and wrap its APIs in the `detail` namespace.

Reply:
Non-public header files for reusable utility code within a specific component is perfectly acceptable. I would not promote it to a `detail` header unless it was useful to other source files outside this directory.

Reply:
I concur with @davidwendt's assessment in general. Headers should go into `include` if they are part of the public API for `libcudf`, and not just because they are headers.

The `TODO` does need removing. (This is a WIP at the moment.) `range_window_bounds` is part of a public API. `<range_window_bounds.hpp>` needs inclusion in `rolling.hpp`.

`range_window_bounds_detail.hpp` is specific to implementation. It shouldn't go into `include/cudf`.